### PR TITLE
FunctionMatcher support for fully qualified function names

### PR DESCRIPTION
**Pull request by lexa-diky**
Now matcher will check both simple name e.g. `foo` and fully qualified one e.g. `com.github.detekt.FooBar.foo`.

Fixes https://github.com/detekt/detekt/issues/5767

### CODE REVIEW

**Comment by BraisGabin** [comment]
This looks good! You didn't relate this with the issue because right now it doesn't fix that issue, right?

**Comment by lexa-diky** [comment]
> This looks good! You didn't relate this with the issue because right now it doesn't fix that issue, right?

I want to go through tests one more time, I have a suspension that they dont check behavior correctly

**Comment by lexa-diky** [comment]
> > This looks good! You didn't relate this with the issue because right now it doesn't fix that issue, right?
> 
> I want to go through tests one more time, I have a suspension that they dont check behavior correctly

Now it should be correct.

Previously most tests assumed that function is in the root package. So if we remove name check inside matcher tests would still pass. Not a case anymore

**Review by BraisGabin — APPROVED** [review]
Good one :) Thanks!

**Comment by BraisGabin** [comment]
I'm re-running CI because we just enabled a rule regarding the identation on the tests (#5819). I think that you need to change something to make the code complaint. Apart from that this is ready to merge :)

File: detekt-tooling/src/test/kotlin/io/github/detekt/tooling/api/FunctionMatcherSpec.kt [hunk 1]
```diff
@@ -261,6 +295,48 @@ class FunctionMatcherSpec(private val env: KotlinCoreEnvironment) {
...
+            result: Boolean
+        ) {
+            val ktFile = compileContentForTest(
+                """
+                package io.github.detekt
+
+                $classSignature {
+                    $functionSignature {}
+                }
+                """.trimIndent()
```
Comment (BraisGabin) [inline detekt-tooling/src/test/kotlin/io/github/detekt/tooling/api/FunctionMatcherSpec.kt:331]:
```suggestion
                """
                    package io.github.detekt

                    $classSignature {
                        $functionSignature {}
                    }
                """.trimIndent()
```

File: detekt-tooling/src/test/kotlin/io/github/detekt/tooling/api/FunctionMatcherSpec.kt [hunk 2]
```diff
@@ -270,8 +346,17 @@ private class TestCase(
...
+private fun buildKtFunction(
+    environment: KotlinCoreEnvironment,
+    code: String,
+    includePackage: Boolean = true
+): Pair<KtNamedFunction, BindingContext> {
+    val ktFile = compileContentForTest(
+        """
+        ${if (includePackage) "package io.github.detekt" else ""}
+        $code
+        """.trimIndent()
```
Comment (BraisGabin) [inline detekt-tooling/src/test/kotlin/io/github/detekt/tooling/api/FunctionMatcherSpec.kt:358]:
```suggestion
        """
            ${if (includePackage) "package io.github.detekt" else ""}
            $code
        """.trimIndent()
```

**Review by cortinico — APPROVED** [review]

**Pull request merged**